test/Microsoft.AspNetCore.Server.KestrelTests/AsciiDecoding.cs

```diff
@@ -150,7 +152,10 @@ private void LargeAllocationProducesCorrectResults()
                 {
                     var returnBlock = block;
                     block = block.Next;
-                    pool.Return(returnBlock);
```

Comment:
Wouldn't it be simpler to keep this as is and remove

                 pool.Return(mem0);
                 pool.Return(mem1);